### PR TITLE
detect-distance: Convert unittests to new FAIL/PASS API

### PR DESCRIPTION
**Pull request by hsadia538**
- [X] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [X] I have signed the Open Information Security Foundation contribution agreement at https://suricata.io/about/contribution-agreement/
- [ ] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/issues/4020) ticket:

Describe changes:

- Update detect-distance unit tests to FAIL/PASS APIs

### CODE REVIEW

**Comment by hsadia538** [comment]
The if (`x==NULL`) conditions had print statements as well which I removed. I was not entirely sure about them but the unit tests FAIL/PASS API does not have any print statements so they are completely skipped.
Please let me know if they need to be there and then I'll look for a way to add them back.

File: src/detect-distance.c [hunk 1]
```diff
@@ -218,25 +194,22 @@ static int DetectDistanceTest01(void)
...
-        goto end;
-
+    FAIL_IF_NULL(p);
     char sig[] = "alert tcp any any -> any any (msg:\"suricata test\"; "
                     "byte_jump:1,2; content:\"|00|\"; "
                     "within:1; distance:2; sid:98711212; rev:1;)";
 
     p->flowflags = FLOW_PKT_ESTABLISHED | FLOW_PKT_TOCLIENT;
-    result = UTHPacketMatchSig(p, sig);
+    UTHPacketMatchSig(p, sig);
```
Comment (victorjulien) [inline src/detect-distance.c:208]: How can the check fail now?

File: src/detect-distance.c [hunk 2]
```diff
@@ -160,55 +160,31 @@ static int DetectDistanceSetup (DetectEngineCtx *de_ctx, Signature *s,
...
     DetectEngineCtx *de_ctx = DetectEngineCtxInit();
-    if (de_ctx == NULL) {
-        printf("no de_ctx: ");
-        goto end;
-    }
+    FAIL_IF_NULL(de_ctx);
 
     de_ctx->flags |= DE_QUIET;
 
     de_ctx->sig_list = SigInit(de_ctx, "alert tcp any any -> any any (content:\"|AA BB|\"; content:\"|CC DD EE FF 00 11 22 33 44 55 66 77 88 99 AA BB CC DD EE|\"; distance: 4; within: 19; sid:1; rev:1;)");
```
Comment (victorjulien) [inline src/detect-distance.c:168]: While changing these tests we would like to change `SigInit` to `DetectEngineAppendSig`. See other tests for how this is different.

**Review by victorjulien — CHANGES_REQUESTED** [review]

File: src/detect-distance.c [hunk 3]
```diff
@@ -218,25 +194,22 @@ static int DetectDistanceTest01(void)
  */
 static int DetectDistanceTestPacket01 (void)
 {
-    int result = 0;
     uint8_t buf[] = { 0x01, 0x00, 0x00, 0x00, 0x00, 0x00 };
     uint16_t buflen = sizeof(buf);
     Packet *p;
     p = UTHBuildPacket((uint8_t *)buf, buflen, IPPROTO_TCP);
```
Comment (jufajardini) [inline src/detect-distance.c:200]: This could become a single statement.

File: src/detect-distance.c (hunk 1 above)
Comment (jufajardini) [reply src/detect-distance.c:208]: You could just directly call the FAIL macro with this call, say:
`FAIL_IF_NOT(UTHPacketMatchSig(p, sig));`

If you do `git grep UTHPacketMatchSig` on the repository, you'll find a few examples.

**Review by jufajardini — COMMENTED** [review]
I've added a few more comments to go with Victor's.

**Comment by hsadia538** [comment]
Followed by #8092

**Pull request closed without merging**